test/unit/kubernetes-deploy/kubernetes_resource/deployment_test.rb

```diff
@@ -307,6 +307,22 @@ def test_deploy_timed_out_based_on_progress_deadline
     end
   end
 
+  def test_deploy_timed_out_based_on_timeout_override
+    Timecop.freeze do
+      template = build_deployment_template
+      template["metadata"]["annotations"][KubernetesDeploy::KubernetesResource::TIMEOUT_OVERRIDE_ANNOTATION] = "10S"
+      deploy = build_synced_deployment(
+        template: template,
+        replica_sets: [build_rs_template(status: { "replica" => 1 })]
+      )
+      refute deploy.deploy_timed_out?, "Deploy not started shouldn't have timed out"
+      deploy.deploy_started_at = Time.now.utc - 3.minutes
```

Comment:
This doesn't necessarily show that the annotation took precedence over the default and the progress deadline. IMO we should add an expired progress condition, an assertion as to what the default is, and an assertion on each edge of the time that the annotation specifies to prove it is the one causing the timeout.

Reply:
I've added assertions on each edge and changed `pretty_timeout_type` to check for timeout override